extension-iap/src/iap_ios.mm

```diff
@@ -477,18 +487,6 @@ static int IAP_SetListener(lua_State* L)
         dmScript::DestroyCallback(iap->m_Listener);
 
     iap->m_Listener = dmScript::CreateCallback(L, 1);
-
-    if (g_IAP.m_Observer == 0) {
-        SKPaymentTransactionObserver* observer = [[SKPaymentTransactionObserver alloc] init];
-        observer.m_IAP = &g_IAP;
-        // NOTE: We add the listener *after* a lua listener is set
-        // The payment queue is persistent and "old" transaction might be processed
-        // from previous session. We call "finishTransaction" when appropriate
-        // for all transaction and we must ensure that the result is delivered to lua.
-        [[SKPaymentQueue defaultQueue] addTransactionObserver: observer];
-        g_IAP.m_Observer = observer;
-    }
-
     return 0;
```

Comment:
This is a behavioral change.
Previously, when when setting the listener, you could receive callbacks.
With this change, it won't happen? That might potentially break our users code I assume.

Please motivate the change, and also if the change should be considered a major number bump.

Reply:
Exactly. You'd get callbacks for any pending purchases when you set the listener. The same behaviour applies to both iOS and Android. I don't think we should change this unless there is a very compelling reason.

Reply:
I see, and I agree with you that it makes sense to do it this way.
Thanks for the explanation.